feat: Update the structure of new Coverage engine release note DOCS-623 (#1970)
codacy/docs: docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md

As a result, both old and new data will coexist during this period, <span class="skip-vale">potentially</span> leading to [differences in reported metrics](#differences-in-coverage-metrics-between-the-old-and-new-coverage-engines). Please refer to the table below for the updated status of the transition process. The table will be updated as changes are introduced.

| Codacy feature             | Transition status | Notes                                                                                                                                        |
|----------------------------|-------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
| Codacy app UI              | Planned           | -                                                                                                                                            |
| Codacy API                 | Planned           | -                                                                                                                                            |

> | Git provider status checks | Beta              | Data from the new Coverage engine is marked \[beta\]<br>and shown alongside data from the current engine. [Read more below](#status-checks). |

Reviewer: Should we order the table by event date? This phase already occurred, so it should be the first line. 🤔

> | Git provider status checks | Beta              | Data from the new Coverage engine is marked \[beta\]<br>and shown alongside data from the current engine. [Read more below](#status-checks). |

Reviewer: I believe we should focus on the main occurrence and leave the details (e.g. data marked beta) to the section below.
Also, what do you think of adding the link in the feature column directly? (I left the original, though)

```suggestion
| [Git provider status checks](#status-checks) | Beta              | The new Coverage engine sends coverage data to your Git provider. This beta data is shown alongside the data from the current engine. [Read more below](#status-checks). |
```

> | GitHub coverage summaries  | Planned           | -                                                                                                                                            |

Author: @codacy/alabama As mentioned during the daily, I'm updating this page to give more visibility to the user what's happening with the transition to the new engine. I'll then update this table as needed as the updates go online.

Let me know if I should add more entries here, if this approach is overkill, or any other observations.

cc @machadoit

Reviewer 2: I noticed that we didn't have a doc task on the summary changes https://codacy.atlassian.net/browse/ALA-589 we are now putting it live as time of writing. 

So we can update the docs to 'Live' on the check? With whatever notes you see suitable. Or if you prefer, feel free to add the task on the epic to update on a different PR.

Overall the changes look pretty cool! 👌

> ## Status checks

Reviewer: My goal was to match the feature name with the one in the table, and make clear that this is the status check from the new engine.
Maybe it's a bit long... Feel free to disregard it. 😉 

```suggestion
## Git provider status checks from the new Coverage engine
```

> !!! info "The following applies to the repositories for which you set Codacy to send pull request coverage status data to your Git provider (see how on [GitHub](../../repositories-configure/integrations/github-integration.md#status-checks), [GitLab](../../repositories-configure/integrations/gitlab-integration.md#pull-request-status), and [Bitbucket](../../repositories-configure/integrations/bitbucket-integration.md#pull-request-status))."

Reviewer: For consistency.

```suggestion
!!! info "This section applies to the repositories for which you set Codacy to send pull request coverage status data to your Git provider (see how on [GitHub](../../repositories-configure/integrations/github-integration.md#status-checks), [GitLab](../../repositories-configure/integrations/gitlab-integration.md#pull-request-status), and [Bitbucket](../../repositories-configure/integrations/bitbucket-integration.md#pull-request-status))."
```
